app/app_utils.cpp

```diff
@@ -1,19 +1,22 @@
 // SPDX-License-Identifier: GPL-2.0-or-later
 
-#include <climits>
+#include "app_utils.hpp"
 #include <cstdlib>
+#include <limits>
 
 namespace Util {
-bool strtol(const char* nptr, long& n) {
+bool strtol(const char* nptr, int64_t& n) {
   if (!nptr || *nptr == '\0')
     return false;
   char* endptr = nullptr;
-  long tmp = std::strtol(nptr, &endptr, 10);
+  long long tmp = std::strtoll(nptr, &endptr, 10);
   if (*endptr != '\0')
     return false;
-  if (tmp == LONG_MAX || tmp == LONG_MIN)
+  if (tmp < std::numeric_limits<int64_t>::min())
```

Comment:
I think you have changed the behavior here. `strtoll` according to cppreference can return the following values:

```
If the converted value falls out of range of corresponding return type, a range error occurs (setting errno to ERANGE) and LONG_MAX, LONG_MIN, LLONG_MAX or LLONG_MIN is returned.
```
`tmp` would be equal to `LLONG_MIN` or `LLONG_MAX` in case of error but not smaller or bigger than such values.

Reply:
Thanks! That's a good catch. I have added this check back.